dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -159,5 +179,161 @@ public void testGetRelatedContentFromMultilingualContent()
+
+            final Structure structure = new StructureTransformer(contentType).asStructure();
+
+            final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+            final ContentletRelationships contentletRelationship = new ContentletRelationships(
+                    contentlet);
+
+            final Relationship relationship = new Relationship(structure, structure,
+                    "parent" + contentType.variable(), "child" + contentType.variable(),
+                    RELATIONSHIP_CARDINALITY.ONE_TO_MANY.ordinal(), false, false);
+
+            final ContentletRelationshipRecords relationshipsRecord = contentletRelationship.new ContentletRelationshipRecords(
+                    relationship,
+                    false);
+
+            contentletRelationship
+                    .setRelationshipsRecords(CollectionsUtils.list(relationshipsRecord));
+
+            final ESIndexAPI esIndexAPI = Mockito.mock(ESIndexAPI.class);
+
+            Mockito.when(esIndexAPI.isClusterInReadOnlyMode()).thenReturn(true);
+
+            assertFalse(
+                    new ESContentletAPIImpl().isCheckInSafe(contentletRelationship, esIndexAPI));
+
+        }finally{
+            if (contentType != null && contentType.id() != null){
+                contentTypeAPI.delete(contentType);
+            }
+        }
+    }
+
+    /**
+     * Test for isCheckInSafe method with relationship fields. It should return true when the cluster
+     * is in read only mode
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void testIsCheckInSafeWithRelationshipsFieldsShouldReturnTrue()
+            throws DotDataException, DotSecurityException {
+        final long time = System.currentTimeMillis();
+        ContentType contentType = null;
+
+        try {
+            contentType = createContentType("test" + time);
+
+            final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+            final ContentletRelationships contentletRelationship = new ContentletRelationships(
+                    contentlet);
+
+            Field field = FieldBuilder.builder(RelationshipField.class).name("newRel")
+                    .contentTypeId(contentType.id()).values(String.valueOf(RELATIONSHIP_CARDINALITY.ONE_TO_MANY.ordinal()))
+                    .relationType(contentType.variable()).build();
+
+            field = fieldAPI.save(field, user);
+
+            final String fullFieldVar = contentType.variable() + StringPool.PERIOD + field.variable();
+
+            final Relationship relationship = relationshipAPI.byTypeValue(fullFieldVar);
+
+            final ContentletRelationshipRecords relationshipsRecord = contentletRelationship.new ContentletRelationshipRecords(
+                    relationship,false);
+
+            contentletRelationship
+                    .setRelationshipsRecords(CollectionsUtils.list(relationshipsRecord));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'CollectionsUtils.list' due to existing static import 'com.dotcms.util.CollectionsUtils.list'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=3712629)